bundle/edu.gemini.wdba.xmlrpc.server/src/main/java/edu/gemini/wdba/tcc/GhostSupport.java

```diff
@@ -0,0 +1,61 @@
+package edu.gemini.wdba.tcc;
+
+import edu.gemini.spModel.gemini.ghost.Ghost;
+
+import java.util.Objects;
+
+public class GhostSupport implements ITccInstrumentSupport {
+    private ObservationEnvironment oe;
+
+    private GhostSupport(final ObservationEnvironment oe) {
+        Objects.requireNonNull(oe, "Observation environment can not be null");
+        this.oe = oe;
+    }
+
+    /**
+     * Factor for creating a new GHOST Instrument Support.
+     *
+     * @param oe The current ObservationEnvironment
+     * @return A new instance
+     * @throws NullPointerException returned if the ObservationEnvironment is null.
+     */
+    static public ITccInstrumentSupport create(final ObservationEnvironment oe) throws NullPointerException {
+        return new GhostSupport(oe);
+    }
+
+    @Override
+    public String getWavelength() {
+        return null;
+    }
+
+    @Override
+    public String getPositionAngle() {
+        Ghost inst = (Ghost) oe.getInstrument();
```

Comment:
We should probably check this assumption (that the instrument is `Ghost`) when we create the `GhostSupport`. Amazingly `ObservationEnvironment` is immutable so it can just be checked once there in construction.

Reply:
Now done in constructor, since `ObservationEnvironment` is immutable.